### ISSUE TITLE
Have "favorite authors" and highlight crates by your favorite authors in crate lists

### ISSUE THREAD

**Issue by carols10cents** [rust-lang/crates.io#494]
In the user research study we did, many people mentioned the reputation of an author as a factor they considered when deciding which crates to try. 

We could do something like show how many other crates an author has, with the assumption that more crates = more experience = better quality, but that might not be true, and publishing lots of empty crates could gain you more reputation defined in this way.

We could distinguish members of any of the Rust teams, but there are lots of experienced people providing high quality crates that would not get this distinction.

Instead, I think it would be useful if each person who uses crates.io could add a crate author to their "favorite" authors, and any search that user does would show an indicator in the list of crates if the crate was written by a favorite author. The list of someone's favorite authors would be private so that each person could decide who they think is experienced, and this feature wouldn't get gamed or become a popularity contest.

**Comment by PritiKumr** [comment]
@carols10cents This is a very cool feature and I'd love to work on this.

**Comment by carols10cents** [comment]
Awesome! Do you have any questions about getting started?

**Comment by PritiKumr** [comment]
Hey @carols10cents 

> I think this should be on crate owners, that is, the users or groups that are in crates.io's database and associated with a crate, not crate author information, that can be anything, read from Cargo.toml

So in terms of the tables and associations, shouldn't we be creating a table, probably called `favorite_users` which will have references to the favorited user and the currentUser?

The other alternative is to associate the user with `crate_owners` records, which is what I was initially thinking. But then a crate owner can be removed from a crate. So this would be not be a good idea IMO.

What do you think?.

**Comment by carols10cents** [comment]
Creating a favorite_users join table sounds right to me!

**Comment by PritiKumr** [comment]
> potentially with what they've been up to lately on crates.io

@carols10cents could you elaborate more on what sort of things would be useful to show here? I was thinking we can show things like new crates, version updates, users / crates they follow. But then this would probably require us to have an activity feed system to implement this effectively, which could be a separate task on its own.

Wondering if we can just list the favorite users in the dashboard for now...

**Comment by carols10cents** [comment]
Yeah, I think just having a list of favorite users on the dashboard (and the ability to unfavorite them there) is a great start!

**Comment by steverob** [comment]
@carols10cents  here are two options regarding the UI indication part of this issue - 

**Highlighted background: **
![2017-07-22_14-39-56_scrot](https://user-images.githubusercontent.com/1220480/28489951-a53a0982-6eec-11e7-96ad-7e3a82f65df4.png)

**Gold crate name**
![2017-07-22_14-38-04_scrot](https://user-images.githubusercontent.com/1220480/28489952-a539ff3c-6eec-11e7-873c-ae3dc55d8166.png)

We could put a message either below or above the list, say - `Highlighted crates are authored by your favorite crate owners` - is that okay?

Another option I've been thinking is to place a star icon along with the text 'Favorite Owner' or just 'Owner'  - so it would be like `[star-icon] Owner` or `[star-icon] Fav Owner`.. something like that maybe?

Looking forward to hearing your and others' thoughts here :)

**Comment by carols10cents** [comment]
I think I like the highlighted background better!

> We could put a message either below or above the list, say - `Highlighted crates are authored by your favorite crate owners` - is that okay?

Sounds great! Let's try the highlighted background option-- my concern with the star icon is that people would expect to be able to click on the star to favorite *crates* and see how many other people have starred it (something people have requested that we don't have plans to do so I don't want to get people's hopes up ;))

**Comment by carols10cents** [comment]
Possible bootstrapping strategy: github api for who you follow on github?

**Comment by steverob** [comment]
Yeah that's a great idea!!!

--
Regards
Steve Robinson

**Comment by BatmanAoD** [comment]
@PritiKumr It looks like you made some progress on this; what was the status of your WIP branch as far as you remember? @steverob did you implement that background-highlighting logic somewhere?

**Comment by BatmanAoD** [comment]
@carols10cents I've revived @PritiKumr's branch on my own fork and gotten it to at least build, but I don't really know what I'm doing with regard to the frontend. Since it's been two years since progress was made, do you think it would be appropriate to restore the help-wanted tag?

**Comment by xkr47** [comment]
Of those two alternatives I like the highlighted background better.. but in the era of **accessibility** I propose to also include a tag e.g. `by ⭐ <author>` in these items, maybe below the downloads couner. This way you would be covered even in case you forgot the purpose of the site-specific hiight.
